cityofaustin/atd-moped
# Store ecapris subproject ID as text instead of number

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/11653

As it turns out, eCapris Subproject IDs often end with a trailing zero, e.g. `12440.010`. It's taken a while for this problem to fully surface, because:
1. Postgres actually lets you store [decimals with a trailing zero](https://www.migops.com/blog/handling-trailing-zeros-with-numeric-datatype-in-postgresql/).
2. The [Socrata data](https://data.austintexas.gov/Transportation-and-Mobility/Transportation-Capital-Project-Financial-Codes/jega-nqf6) we use to connect projects to funding numbers is sourced from a database view which also had the trailing zeroes truncated. (That has since been corrected effective March 8 2023).

This PR:
1. Changes the `ecapris_subproject_Id` column from `numeric` to `text`. Postgres kindly handles the re-casting automatically. 
2. Fixes existing records in the DB which are missing their trailing zero.
3. Drops the `capitally_funded` column, which we were not actually using.
4. Updates the editor code to properly handle `ecapris_subproject_id` as text.


## Testing
**URL to test:** Local - ideally with a copy of production.

**Steps to test:**
1. Create a new project
2. From the summary tab, set the **eCAPRIS Subproject ID** field to `12440.010`
3. Refresh the page. Observe that the subproject ID correctly renders with the trailing zero.
4. Try editing the subproject ID to have four digits after the decimal and/or a letter. You should get an error:
```
Invalid eCapris value: 12440.0100 must not contain letters and have exactly three digits after the decimal place. E.g., 12680.010.
```
5. Navigate to the "Funding" tab. Add a new funding source by clicking **Add Funding Source** and then select the**From eCapris** dropdown item.
6. In the modal that appears, you will see the funding source for **Pleasant Valley at Lighthouse PHB**. Add it to the project.

You did it!

---
#### Ship list
